DependencyInjection/MobileDetectExtension.php

```diff
@@ -49,6 +49,7 @@ public function load(array $configs, ContainerBuilder $container)
 
         $container->setParameter('mobile_detect.mobile_detector.class', $config['mobile_detector_class']);
         $container->setParameter('mobile_detect.device_view.class', $config['device_view_class']);
+        $container->setParameter('mobile_detect.data_collector_device.class', $config['data_collector_device_class']);
```

Comment:
I would not add a parameter for this (it's one extension point that strictly taken must be maintained). By the way, I don't see where you actually register the service as a data collector.